cmake/modules/FindAVIF.cmake

```diff
@@ -0,0 +1,92 @@
+# - Find the native AVIF includes and library
+#
+
+# This module defines
+#  AVIF_INCLUDE_DIR, where to find avif.h
+#  AVIF_LIBRARIES, the libraries to link against to use AVIF.
+#  AVIF_FOUND, If false, do not try to use AVIF.
+# also defined, but not for general use are
+#  AVIF_LIBRARY, where to find the AVIF library.
+#
+# AVIF needs an encoder. Several are available.
+# You can choose an encoder by setting AVIF_ENCODER.
+# Presently this is set to "aom", a standard encoder.
+# This module also defines AVIF_ENCODER_LIBRARY
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are
+# met:
+#
+#  * Redistributions of source code must retain the above copyright notice,
+#    this list of conditions and the following disclaimer.
+#
+#  * Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the following disclaimer in the documentation
+#    and/or other materials provided with the distribution.
+#
+#  * The names of Kitware, Inc., the Insight Consortium, or the names of
+#    any consortium members, or of any contributors, may not be used to
+#    endorse or promote products derived from this software without
+#    specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDER AND CONTRIBUTORS ``AS IS''
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE AUTHORS OR CONTRIBUTORS BE LIABLE FOR
+# ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
```

Comment:
this should be checking pkg-config first.  look at how other modules use `PkgConfig`.

Reply:
The file `cmake/modules/FindAVIF.cmake` should be removed. libavif ships a cmake config file (something like pkg-config but for cmake!)

Reply:
I'm afraid I'll need guidance from you here, as I'm newer to this build process. I modeled this module after `FindWEBP.cmake`. I thought the build would need to know where to find libavif's header files - and to make sure they existed. Then we can set `AVIF_FOUND` and `HAVE_LIBAVIF`. So, the way it is now mirrors how other resources are discovered.

If I remove this file, I suppose we'll be assuming that libavif exists as long as the developer builds with `-DENABLE_AVIF=1` and `config.h.cmake` contains `#cmakedefine HAVE_LIBAVIF`?

Reply:
@cryptomilk, your next comment gave me a hint. `find_package()` can be used to grab settings from an external project.

Reply:
If you have an option to turn on building with AVIF support, then you should require the package:

```
if (ENABLE_AVIF)
    find_package(libavif 0.8.2 CONFIG REQUIRED)
endif()
```

If you want automatic detection you would go with:

```
find_package(libavif 0.8.2 CONFIG)
if (TARGET avif)
    # build avif support
endif()
```

The header file information is part of the avif target ...

Reply:
we must always have a knob that allows the user to explicitly enable or disable a feature.  it's OK for the default to test & enable the feature if it's available, but there still has to be a manual control knob.

if you look at the configure script, it does this for all the libs we support.

Reply:
excellent! I'd already implemented the first option above: `if (ENABLE_AVIF)` etc.

Of course, now the build can't find the config file. My libavif build placed it in `libavif/CMakeFiles/Export/lib/cmake/libavif/libavif-config.cmake`. Should I have instructed libavif to put export files in a shared location? If so, how would I do that?

Reply:
based on earlier changes, i'm guessing you haven't actually installed libavif, just built it locally.  if you install the library, the cmake file should end up in a path like `/usr/lib64/cmake/` or `/usr/share/cmake/Modules/`.  then cmake can find it.

Reply:
Yep, that's it - the build instructions for libavif I was reading included the flag `-DBUILD_SHARED_LIBS=OFF`. Just switched that out, and I'm rebuilding.

Reply:
Finally realized I had to do a `make install` 😭